packages/odyssey-react-mui/src/components/PasswordInput/PasswordInput.tsx

```diff
@@ -0,0 +1,111 @@
+/*!
+ * Copyright (c) 2022-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import React, { forwardRef, useState, useMemo } from "react";
+import type { Ref, ChangeEvent, MouseEvent } from "react";
+import type { OutlinedInputProps, TooltipProps } from "@mui/material";
+import {
+  Tooltip,
+  IconButton,
+  Box,
+  InputLabel,
+  OutlinedInput,
+  InputAdornment,
+} from "@mui/material";
+import { Visibility, VisibilityOff } from "@mui/icons-material";
+import { useOid } from "../../utils";
+
+interface State {
+  password: string;
+  showPassword: boolean;
+}
+
+export interface PasswordInputProps
+  extends Omit<OutlinedInputProps, "type" | "label" | "defaultValue" | "ref"> {
+  ref?: Ref<HTMLInputElement>;
+  defaultValue?: string;
+  label: string;
+  tooltipLabel?:
+    | TooltipProps["title"]
+    | ((isHidden: boolean) => TooltipProps["title"]);
+}
+
+export const PasswordInput = forwardRef<HTMLInputElement, PasswordInputProps>(
+  (props, ref) => {
+    const {
+      tooltipLabel,
+      id,
+      label,
+      defaultValue: password = "",
+      inputProps,
+      ...rest
+    } = props;
+
+    const [values, setValues] = useState<State>({
+      password,
+      showPassword: false,
+    });
+
+    const handlePasswordChange = (event: ChangeEvent<HTMLInputElement>) => {
+      setValues({ ...values, password: event.target.value });
+      props.onChange?.(event);
+    };
+
+    const handleClickShowPassword = () => {
+      setValues({
+        ...values,
+        showPassword: !values.showPassword,
+      });
+    };
+
+    const handleMouseDownPassword = (event: MouseEvent<HTMLButtonElement>) => {
+      event.preventDefault();
+    };
```

Comment:
In the case of `handleMouseDownPassword`, this _is_ a click handler because it cares about mouse events. I was thinking `onMouseDown`. It could be more specific like `onInputMouseDown`. In terms of the types, does it _have_ to be a mouse event on a button? All this cares about is that it's an `Event`, but a type of event that allows preventing the default behavior. Even if it's a `MouseEvent`, we don't need to specify it as an `HTMLButtonElement`, we could simply have it be `HTMLElement` if that's possible. The `preventDefault` function works the same on any HTML element. The way it's written right now; we're tightly coupling this to `button` elements.